docs/http/http-metrics.md

```diff
@@ -75,18 +75,24 @@ of `[ 0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5,
 |---|---|---|---|---|
 | `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Conditionally Required: If and only if it's available |
-| `http.request.method` | string | HTTP request method. [2] | `GET`; `POST`; `HEAD` | Required |
+| `http.error.reason` | string | Low-cardinality error reason. [2] | `timeout`; `name_resolution_error`; `server_certificate_invalid` | Conditionally Required: [3] |
+| `http.request.method` | string | HTTP request method. [4] | `GET`; `POST`; `HEAD` | Required |
 | `http.response.status_code` | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: If and only if one was received/sent. |
 | [`network.protocol.name`](../general/attributes.md) | string | [OSI Application Layer](https://osi-model.com/application-layer/) or non-OSI equivalent. The value SHOULD be normalized to lowercase. | `amqp`; `http`; `mqtt` | Recommended |
-| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [3] | `3.1.1` | Recommended |
-| [`server.address`](../general/attributes.md) | string | Name of the local HTTP server that received the request. [4] | `example.com` | Opt-In |
-| [`server.port`](../general/attributes.md) | int | Port of the local HTTP server that received the request. [5] | `80`; `8080`; `443` | Opt-In |
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [5] | `3.1.1` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Name of the local HTTP server that received the request. [6] | `example.com` | Opt-In |
+| [`server.port`](../general/attributes.md) | int | Port of the local HTTP server that received the request. [7] | `80`; `8080`; `443` | Opt-In |
 | [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https` | Required |
 
 **[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
 SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
 
-**[2]:** HTTP request method value SHOULD be "known" to the instrumentation.
+**[2]:** Instrumentations SHOULD define and document a list of applicable error reasons that describe why request has failed or response was not received partially or fully. For example, request could be canceled, DNS resolution or TLS handshake may fail, or the connection maybe prematurely terminated before response is sent/received.
+Error reason values MUST be a closed set known at the development time of the instrumentation and MUST have low cardinality.
```

Comment:
Even though each particular instrumentation might produce low cardinality values on its own, if there are 20 different HTTP clients (1-4 per each programming language) used in an organization and thus 20 different HTTP client instrumentations, each defining its own list of values will still result in high cardinality. If there are ~15 error conditions each, there will be 300 different values being sent the same metrics backend.

Reply:
assuming an application uses 20 different HTTP clients for 20+ different destinations (using different HTTP clients for the same destination is probably uncommon), then each destination will have ~15 error codes regardless whether they are unified or not.

E.g. 
- I call `server.address: b` from  `service.name: a` with Java `URLConnection` and sometimes get 
  -  `http.error.reason: timeout`
  -  `http.error.reason: java.io.IOException`
- I call `server.address: c` from  `service.name: a` with HttpClient  `foo` and sometimes  get 
  -  `http.error.reason: bar`
  -  `http.error.reason: baz`

there are 4 time series here regardless what `a`->`c` call reports in `http.error.reason`.

Having custom values only affects the case when people use multiple clients for the same destination in the same sevice. 

In practice, each HTTP client would report a few possible errors reasons (e.g. timeout/cancellation, connection reset, dns, tls, rate-limiting, circuit-breaking) in some form. theoretical cadinality with esoteric or app-specific errors is higher.

So, I'd say it's a perfect user experience, but I don't see a cadinality problem here.

Reply:
discussed at SIG meeting:
- there is no cardinality issue on the application that affects number of time series, memory aggregation or metrics storage directly
- but telemetry consumers should expect high query-time cardinality when not applying source/destination filtering. Added a note on this.

Reply:
also added a note on the attribute that consumer may get high-ish cardinality at a query time (without other filters)